api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -89,7 +95,11 @@ public static Tagger getTagger() {
    * @throws IllegalStateException if a specified provider is not found
    */
   @Nullable
-  private static <T> T loadSpi(Class<T> providerClass) {
+  private static <T> T loadSpi(Class<T> providerClass, ClassLoader classLoader) {
+    if (classLoader.getParent() != null) {
+      return loadSpi(providerClass, classLoader.getParent());
```

Comment:
Please add a comment why you're making a recursive call.  Why do you need the root classloader?

Reply:
+1 will update once we can confirm this works well in multi deployment app servers.